### ISSUE TITLE
how to detect bouncing ball coordinates from on realtime playing

### ISSUE THREAD

**Issue by Razamalik4497** [ultralytics/yolov5#9529]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

![image](https://user-images.githubusercontent.com/49755089/191504107-d154b353-1aa5-4a99-b572-5b1679ac2d05.png)

How can I find the bounced ball coordinates?
I'm working with yolov5 the ball is detecting on the Realtime with 120 Fps please see the attached image and help me out what can be possible way to detect the boucing ball cordinates

**Comment by glenn-jocher** [comment]
👋 Hello! Thanks for asking about **handling inference results**. YOLOv5 🚀 [PyTorch Hub](https://docs.ultralytics.com/yolov5/tutorials/pytorch_hub_model_loading) models allow for simple model loading and inference in a pure python environment without using `detect.py`. 

### Simple Inference Example

This example loads a pretrained YOLOv5s model from PyTorch Hub as `model` and passes an image for inference. `'yolov5s'` is the YOLOv5 'small' model. For details on all available models please see the [README](https://github.com/ultralytics/yolov5#pretrained-checkpoints). Custom models can also be loaded, including custom trained PyTorch models and their [exported](https://docs.ultralytics.com/yolov5/tutorials/model_export) variants, i.e. ONNX, TensorRT, TensorFlow, OpenVINO YOLOv5 models.

```python
import torch

# Model
model = torch.hub.load('ultralytics/yolov5', 'yolov5s')  # yolov5n - yolov5x6 official model
#                                            'custom', 'path/to/best.pt')  # custom model

# Images
im = 'https://ultralytics.com/images/zidane.jpg'  # or file, Path, URL, PIL, OpenCV, numpy, list

# Inference
results = model(im)

# Results
results.print()  # or .show(), .save(), .crop(), .pandas(), etc.
results.xyxy[0]  # im predictions (tensor)

results.pandas().xyxy[0]  # im predictions (pandas)
#      xmin    ymin    xmax   ymax  confidence  class    name
# 0  749.50   43.50  1148.0  704.5    0.874023      0  person
# 2  114.75  195.75  1095.0  708.0    0.624512      0  person
# 3  986.00  304.00  1028.0  420.0    0.286865     27     tie

results.pandas().xyxy[0].value_counts('name')  # class counts (pandas)
# person    2
# tie       1
```
<img width="550" src="https://user-images.githubusercontent.com/26833433/149679662-b2021c78-59b2-4f52-b768-1e5161537c85.png">

See [YOLOv5 PyTorch Hub Tutorial](https://docs.ultralytics.com/yolov5/tutorials/pytorch_hub_model_loading) for details.

Good luck 🍀 and let us know if you have any other questions!

**Comment by barney2074** [comment]
you mean the pixel coordinates ? (as opposed to real world cartesian XYZ coordinates)

detect.py has normalised coordinates of the centre of the bounding box:
https://github.com/ultralytics/yolov5/blob/489920ab30b217fed14d3ddd31c23e9afc5be238/detect.py#L159

**Comment by Zephyr69** [comment]
The coordinates are earily found in the code. But the challenge here is a clearly annotated dataset, which also implies a highly capable imaging system. I don't think mere 120fps will allow you to catch a clear image of the flying ball.

**Comment by Razamalik4497** [comment]
> The coordinates are earily found in the code. But the challenge here is a clearly annotated dataset, which also implies a highly capable imaging system with extremely short exposure time. I don't think mere 120fps (if retrieving raw live image) will allow you to catch a clear image of the flying ball.

I have ball coordinates detected but now how to find the "bouncing ball" coordinates only, I think it can be done with the help of advance logics? what you recommend

**Comment by glenn-jocher** [comment]
@barney2074 understood! Detecting the bouncing ball can indeed be a challenging task due to the high-speed nature of the movement. To specifically identify the bouncing ball, you may need to implement additional logic to track its trajectory over consecutive frames and detect the pattern of bouncing. Since YOLOv5 detects objects in individual frames, you may need to incorporate external image processing or computer vision techniques for tracking and analyzing the ball's movement across frames. Additionally, integrating motion analysis algorithms or optical flow methods could provide insights into the ball's bouncing pattern. Good luck with your project, and feel free to reach out if you have further questions!
